src/components/Utils/firestore.js

```diff
@@ -10,3 +10,13 @@ export const getProducts = () => {
   const token = localStorage.getItem('tcl18-token');
   return db.collection(token);
 };
+
+export const existCollectionByToken = async (userToken) => {
+  const collectionRefByToken = db.collection(userToken || ' ');
+  try {
+    const response = await collectionRefByToken.get();
+    return !response.empty;
+  } catch (e) {
+    return false;
+  }
```

Comment:
I'm glad you're taking care of handling errors. 👏 

**Sidenote.** In other more complex projects, you wouldn't want to `return false` in the `catch` block (this is called error swallowing). In such projects you'd want to:

- Report the error you just got your error reporting mechanisms like [Sentry](https://sentry.io/welcome/)
- Let the `error` bubble up so you see the real error in your console. You'll always want to know the **real** error so you can fix it. Getting `false` means nothing 😅

Reply:
In a more complex project, it could look like this

```js
try {
  // your optimistic code
} catch (error) {
  reportToSentry(error);
  // because you're not returning or throwing anything, the error will bubble up.
}
```

Reply:
Thanks, @alejandronanez, excellent observation 💯👀 ! 

Well, we put `console.err(error)` at least now, we will have the opportunity to see errors in the console 😄😅 🥳.

(PS. Thanks, I'll research a little about `sentry`, maybe I can learn a little about it. I'm so sorry, I didn't know about it and I can see It's very useful 😄)

Reply:
>  I'm so sorry, I didn't know about it and I can see It's very useful

Don't be it's hard to stay on top of everything these days.